test/solvers/functional/functional_test.py

```diff
@@ -199,15 +197,21 @@ def test_right_scalar_multiplication(space):
     assert isinstance((F * 0), odl.solvers.ConstantFunctional)
     assert almost_equal((F * 0)(x), F(space.zero()), places=PLACES)
 
-    # TODO: Add more test for convex conjugate? Old ConvexConjugateFuncScaling?
-
-    # Test proximal operator. This requiers scaling to be positive.
+    # Test proximal operator: positive, negative and zero scalar
     sigma = np.random.rand()
     assert all_almost_equal(((F * scal).proximal(sigma))(x),
-                            ((1.0 / scal) *
-                                (F.proximal(sigma * scal**2)))(x * scal),
+                            ((1.0 / scal) * F.proximal(
+                                sigma * scal**2))(x * scal),
                             places=PLACES)
 
+    assert all_almost_equal(((F * neg_scal).proximal(sigma))(x),
+                            ((1.0 / neg_scal) * F.proximal(
+                                sigma * neg_scal**2))(x * neg_scal),
+                            places=PLACES)
+
+    # The proximal operator of a constant functional is the identity operator
+    assert isinstance((F * 0).proximal(sigma), odl.IdentityOperator)
```

Comment:
This test should basically only concern right multiplication. On line 161 in the same file, in the test for left multiplication:
```python
assert isinstance(0 * F, odl.solvers.ZeroFunctional)
```
The rest should follow from that, as long as the `ZeroFunctional` pass the tests :)